src/analyses/base.ml

```diff
@@ -1127,6 +1127,7 @@ struct
       | `Bot   -> Queries.ID.top () (* out-of-scope variables cause bot, but query result should then be unknown *)
       | `Top   -> Queries.ID.top () (* some float computations cause top (57-float/01-base), but query result should then be unknown *)
       | v      -> M.debug ~category:Analyzer "Base EvalInt %a query answering bot instead of %a" d_exp e VD.pretty v; Queries.ID.bot ()
+      | exception (IntDomain.ArithmeticOnIntegerBot _) when not !AnalysisState.should_warn -> Queries.ID.top () (* for some privatizations, values can intermediately be bot because side-effects have not happened yet *)
```

Comment:
Why conditional on `should_warn` though? In `eval_rv` (which this bypasses) it's unconditional.

Reply:
The idea was to let this still fail if it happens when a solution has been computed, since I am not sure if it is also benign if it happens there.